### DISCUSSION TITLE
customize breadcrumbs

### DISCUSSION THREAD

**Discussion by sean-au** [bootscore/bootscore#272] [Q&A]
Hi, 

I have spent a fair amount of time digging into this theme and its great!

I am wondering if anyone could help me on how to customize the breadcrumbs, I can't get my head around it. 

I have tried copying and pasting the whole block from functions.php into a child theme and removing the first and last lines so the following is in the child theme:

```
function the_breadcrumb() {
    if (!is_home()) {
      echo '<nav class="breadcrumb mb-4 mt-2 bg-light py-2 px-3 small rounded">';
      echo '<a href="' . home_url('/') . '">' . ('<i class="fa-solid fa-house"></i>') . '</a><span class="divider">&nbsp;/&nbsp;</span>';
      if (is_category() || is_single()) {
        the_category(' <span class="divider">&nbsp;/&nbsp;</span> ');
        if (is_single()) {
          echo ' <span class="divider">&nbsp;/&nbsp;</span> ';
          the_title();
        }
      } elseif (is_page()) {
        echo the_title();
      }
      echo '</nav>';
    }
  }
  add_filter('breadcrumbs', 'breadcrumbs');
```

But it won't override the breadcrumbs. 

I have also tried looking through the theme to figure out how its hooked in, but I cant figure that out either. 

Thanks in advance!

**Comment by crftwrk** [comment]
Hi Sean,

copy the breadcrumb function to your child is the correct way. At a first glance to your code, you didn't change anything? That's why you see no changes?

Try this:

```php
function the_breadcrumb() {
  if (!is_home()) {
    echo '<nav class="breadcrumb mb-4 mt-2 bg-warning text-white py-2 px-3">';
    echo '<a href="' . home_url('/') . '">' . ('Home') . '</a><span class="divider text-success">&nbsp;»&nbsp;</span>';
    if (is_category() || is_single()) {
      the_category(' <span class="divider text-success">&nbsp;»&nbsp;</span> ');
      if (is_single()) {
        echo ' <span class="divider text-success">&nbsp;»&nbsp;</span> ';
        the_title();
      }
    } elseif (is_page()) {
      echo the_title();
    }
    echo '</nav>';
  }
}
add_filter('breadcrumbs', 'breadcrumbs');
```

### Output

![screenshot](https://user-images.githubusercontent.com/51531217/193231646-124361de-27e6-4abe-8c24-e4c4ccda0b93.png)

https://dev.bootscore.me/2012/01/07/template-sticky/

Does this help?

**Comment by sean-au** [comment]
Hi Basti,

I just copied the original code to make it simple.. I wasn't able to debug
my code as it would only display the original.

I'll try again and get back to you.


Also how does the final line work with add filter?

**Reply by sean-au** [reply]
Thanks again for getting back to me. 

I had another go and realized that there is a separate function for woocommerce breadcrumbs. 

I'll have a go at setting them both up to work with bootstrap 5.2 breadcrumb spec and submit a PR.

**Reply by crftwrk** [reply]
Yes, forgot. WooCommerce breadcrumb uses a different function.

```php
add_filter('woocommerce_breadcrumb_defaults', 'bs_woocommerce_breadcrumbs');
function bs_woocommerce_breadcrumbs() {
  return array(
    'delimiter'   => ' &nbsp;&#47;&nbsp; ',
    'wrap_before' => '<nav class="breadcrumb mb-4 mt-2 bg-light py-2 px-3 small rounded" itemprop="breadcrumb">',
    'wrap_after'  => '</nav>',
    'before'      => '',
    'after'       => '',
    'home'        => _x('Home', 'breadcrumb', 'woocommerce'),
  );
}
```

You closed https://github.com/bootscore/bootscore/pull/273?

**Reply by sean-au** [reply]
PR submitted
